taskcluster/utils.py

```diff
@@ -405,3 +406,11 @@ def do_GET(self):
     while not creds[0]:
         server.handle_request()
     return creds[0]
+
+
+def singleTrailingSlash(url):
+    return '{}/'.format(url.rstrip('/'))
+
+
+def joinBaseUrlAndRoute(baseUrl, route):
```

Comment:
If we did the singleTrailingSlash logic in this function, we would be safe from the concern about the baseurl changing after instantiation.

Also, since this is logic which is targetting client stuff specifically, what about making it available as BaseClient._joinBaseUrlAndRoute instead?

Reply:
Sure. I'll go ahead and move the code.